pvlib/pvsystem.py

```diff
@@ -432,20 +434,39 @@ def sapm(self, effective_irradiance, temp_cell, **kwargs):
         """
         return sapm(effective_irradiance, temp_cell, self.module_parameters)
 
-    def sapm_celltemp(self, irrad, wind, temp):
-        """Uses :py:func:`sapm_celltemp` to calculate module and cell
-        temperatures based on ``self.racking_model`` and
-        the input parameters.
+    def sapm_celltemp(self, poa_global, temp_air, wind_speed,
+                      parameter_set=None):
+        """Uses :py:func:`temperature.sapm_cell` to calculate cell
+        temperatures.
 
         Parameters
         ----------
-        See pvsystem.sapm_celltemp for details
+        poa_global : numeric or Series
+            Total incident irradiance in W/m^2.
+
+        temp_air : numeric or Series
+            Ambient dry bulb temperature in degrees C.
+
+        wind_speed : numeric or Series
+            Wind speed in m/s at a height of 10 meters.
+
+        parameter_set : string, default None
```

Comment:
Instead of a kwarg, maybe we should infer from `module_type` and `racking_model`? That could give us the default behavior back in a clean way (and is more consistent with the `PVSystem` philosophy of set the attributes, then run the models). Not sure what the order of operations should be. Maybe first check `temperature_model_parameters`, then try `racking_model + module_type`?

Reply:
I guess this doesn't quite give us the default behavior at the `ModelChain` layer. That same kind of look up would be needed in `infer_temperature_model`.

Reply:
> Instead of a kwarg, maybe we should infer from module_type and racking_model? That could give us the default behavior back in a clean way (and is more consistent with the PVSystem philosophy of set the attributes, then run the models). Not sure what the order of operations should be. Maybe first check temperature_model_parameters, then try racking_model + module_type?

That's the long term plan, but right now we don't have a source of values for `module_type`, which should accompany the module parameters. I hope we can add that data as part of the SAM file build.

Reply:
Looks like `parameter_set` can be removed. `PVSystem` attributes should contain this kind of metadata. `PVSystem` methods should only accept the time-varying weather data.